lib/oli/delivery/distributed_depot_coordinator.ex

```diff
@@ -0,0 +1,47 @@
+defmodule Oli.Delivery.DistributedDepotCoordinator do
+  @moduledoc """
+  This module is responsible for coordinating updates to the depot across the cluster.
+  """
+
+  use GenServer
+
+  alias Phoenix.PubSub
+  alias Oli.Delivery.Depot
+  alias Oli.Delivery.Depot.DepotDesc
+
+  def start_link(init_args),
+    do: GenServer.start_link(__MODULE__, init_args, name: __MODULE__)
+
+  def update_all(%DepotDesc{} = depot_desc, entries),
+    do: PubSub.broadcast(Oli.PubSub, topic(), {:update_all, depot_desc, entries})
+
+  def clear(%DepotDesc{} = depot_desc, table_id),
+    do: PubSub.broadcast(Oli.PubSub, topic(), {:clear, depot_desc, table_id})
+
+  def init(_) do
+    PubSub.subscribe(Oli.PubSub, topic())
+    {:ok, []}
+  end
+
+  def handle_info({:clear, depot_desc, table_id}, state) do
+    if Depot.table_exists?(depot_desc, table_id) do
+      Depot.clear(depot_desc, table_id)
+    end
+
+    {:noreply, state}
+  end
+
+  def handle_info({:update_all, depot_desc, entries}, state) do
+    [first | _rest] = entries
+    table_id = Map.get(first, depot_desc.table_id_field)
+
+    if Depot.table_exists?(depot_desc, table_id) do
```

Comment:
If the table does not exist, shouldn't we create and populate it? Or do we only want to initialize the table when we explicitly need the data (when calling any of the SectionResourceDepot functions)?